src/domain/entities/User.ts

```diff
@@ -0,0 +1,61 @@
+import { Id } from '../value-objects/Id';
+import { Name } from '../value-objects/Name';
+import { Email } from '../value-objects/Email';
+import { Image } from '../value-objects/Image';
+import { CreatedAt } from '../value-objects/CreatedAt';
+import { UpdatedAt } from '../value-objects/UpdatedAt';
+
+export interface UserProps {
```

Comment:
Para exponer las props y no perder la encapsulación, seria mejor usar un metodo `toPrimitives()`
```suggestion
interface UserProps {
```

Reply:
La encapsulación no se rompe por exportar una interfaz. Lo que la rompe es permitir acceso o mutación sin control, y eso acá no ocurre. Los props siguen siendo privados, no hay setters públicos, y el único punto de entrada sigue siendo el constructor. Exportar UserProps simplemente me permite tipar correctamente lo que entra por el constructor de la entidad.
En cuanto a toPrimitives() y otros métodos de la misma linea como  fromPrimitives() no son obligatorios. Puedes construir una instancia directamente con el constructor y acceder a sus datos con los getters sin problema.

Colocar esos métodos es simplemente gustos de cada desarrollador, repito no son obligatorios.

Reply:
Con a romper la encapsulación me refiero a que estas exponiendo la estructura interna(props) de la entidad a capas externas. Los metodos `fromPrimitves` y `toPrimitives` te permite no exponer la estructura interna (props) y defines un contrato de serialización y deserialización más controlado.

Reply:
Por ejemplo desde cualquier capa exterior tendría que usar un VO para crear un usuario. 
```ts
const user = new User({
  id: new Id('1'),
  name: new Name('Anon'),
  ...
});
```